mlflow/utils/validation.py

```diff
@@ -36,6 +44,7 @@ def _validate_metric_name(name):
         raise Exception("Invalid metric name: '%s'. %s" % (name, _BAD_CHARACTERS_MESSAGE))
     if path_not_unique(name):
         raise Exception("Invalid metric name: '%s'. %s" % (name, bad_path_message(name)))
+    _validate_length_limit("Metric name", MAX_ENTITY_KEY_LENGTH, name)
```

Comment:
Open question (copied from [this comment](https://github.com/mlflow/mlflow/pull/950#discussion_r262748862) on #950): do we want to standardize metric/param/tag name & value length validations in OSS by adding such validation logic here?

For context, we currently validate metric/param/tag names in the user-facing Python SDK (fluent & client APIs) as well as in the backing store implementations (SqlAlchemyStore, FileStore). However, length limits are currently defined per-store (i.e. the SQLAlchemyStore restricts metric names to 250 chars, while the FileStore imposes no limit but will fail to write metrics if their names exceed the filesystem's max filename length).

It seems most consistent to have the same length validations across stores so that the behavior/validity of user code isn't dependent (or is minimally dependent) on what store they're running against, so I added such validation logic here - it is a breaking behavior change though, so it's worth debating

Reply:
Thinking about this, it's probably best not to break FileStore workflows that log big params / it's probably best to just apply this validation to the inputs of `log_batch` (which is an entirely new API)